### ISSUE TITLE
Dialyzer warnings

### ISSUE THREAD

**Issue by cjbottaro** [zhongwencool/eetcd#56]
```
The call eetcd_lock:lock
         (_ctx@1 ::
              #{'eetcd_reply_timeout' := 'infinity' | integer(), _ => _},
          _key@1 :: any(),
          _lease_id@1 :: any()) will never return since it differs in the 1st argument from the success typing arguments: 
         (atom() | reference(),
          binary(),
          pos_integer())ElixirLS Dialyzer
```

This is how I'm calling `lock/3`:
```elixir
ctx = :eetcd_lock.new(Datastores.Etcd)
|> :eetcd_lock.with_timeout(5_000)

:eetcd_lock.lock(ctx, "foo", lease_id)
```

Where `lease_id` is a positive integer.

This only happens for `eetcd_lock`, other modules like `eetcd_lease` work fine with a map as `ctx`.

I tried looking at the source code, but then remembered that I don't know Erlang. That said, `eetc_lease` seems the same as `eetcd_lock` in terms of using `Context`, so not sure why Dialyzer doesn't like it in the latter.

**Comment by belltoy** [comment]
It should be fixed in #57.

**Comment by cjbottaro** [comment]
Yep, it does! Can you merge it and cut a release? 🙏 cc @zhongwencool
